pkg/util/hostname/providers.go

```diff
@@ -185,6 +206,20 @@ func GetWithProvider(ctx context.Context) (Data, error) {
 	return Data{}, err
 }
 
+// GetWithProviderWithoutCacheAndIMDSV2 returns the hostname for the Agent and the provider that was use to retrieve it without using the cache and without using IMDSv2
+func GetWithProviderWithoutCacheAndIMDSV2(ctx context.Context) (Data, error) {
```

Comment:
First, I think it makes sense to use cache for the new field. Also having a cache or not is implementation specific and shouldn't be reflected in the naming of the function when a cache is not used. What do you think?